hudi-common/src/main/java/org/apache/hudi/common/model/HoodieAvroRecord.java

```diff
@@ -184,16 +185,20 @@ public HoodieRecord<T> copy() {
 
   @Override
   public HoodieRecord wrapIntoHoodieRecordPayloadWithParams(
-      Schema recordSchema, Properties props,
+      Schema recordSchema,
+      Properties props,
       Option<Pair<String, String>> simpleKeyGenFieldsOpt,
       Boolean withOperation,
       Option<String> partitionNameOp,
       Boolean populateMetaFields,
       Option<Schema> schemaWithoutMetaFields) throws IOException {
-    IndexedRecord indexedRecord = (IndexedRecord) data.getInsertValue(recordSchema, props).get();
+    HoodieAvroRecordMerger merger = HoodieAvroRecordMerger.INSTANCE;
```

Comment:
Can we inline the variable `merger` ? If we want to remove all the usage of `getInsertValue`, we should deprecate it first.

Reply:
Hi Danny, what do you mean by "inline"? Do you mean we should pass the merger from the parameters? Meanwhile, how do we deprecate this function?

Reply:
Hi Danny, after discussion with @yihua, I will create a separate task to deprecate these APIs for HoodieAvroRecord class if possible for now.